modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -67,7 +73,7 @@ def execute cmd, msg=nil, error_only=false
         std_err.close unless std_in.nil?
       end
       report msg, response, error_only
-      response
+      response if return_result
```

Comment:
Oops, of course. Still, I don't like the idea of a conditional return; let's explicitly return nils from  create_* and remove_* methods instead.